src/main/java/app/fitbuddy/annotation/validator/FitBuddyDateValidator.java

```diff
@@ -0,0 +1,35 @@
+package app.fitbuddy.annotation.validator;
+
+import app.fitbuddy.annotation.FitBuddyDate;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+@Service
+public class FitBuddyDateValidator implements ConstraintValidator<FitBuddyDate, String> {
+
+	private final String datePattern;
+
+	@Autowired
+	public FitBuddyDateValidator(@Value("${fitbuddy.validation.pattern.date:yyyy-MM-dd}") String datePattern) {
+		this.datePattern = datePattern;
+	}
+
+	@Override
+	public boolean isValid(String value, ConstraintValidatorContext context) {
+		DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(datePattern);
+		try {
+			LocalDate.parse(value, dateTimeFormatter);
+			return true;
+		} catch (Throwable ignoredThrowable) {
```

Comment:
I would only catch `DateTimeParseException` here in Line 31.

This is a code smell: [sonar cloud](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=166&id=fitbuddy-app&open=AYTHDoWKoP7WDH0sL0MS)